ansible/playbooks/restore-backup.yml

```diff
@@ -0,0 +1,106 @@
+---
+# This playbook can be run on staging to test the integrity of the backup, and
+# in production to set up a new production environment. Because it always needs
+# access to the production AWS bucket, when running on staging you need to add
+# "--vault-id production@prompt" to the run-playbook.sh call!
+- hosts: "all"
+  user: "ansible"
+  become: true
+  become_user: "root"
+  become_method: "sudo"
+  force_handlers: true
+
+  vars_prompt:
+    - name: "confirm"
+      default: 'ABORT'
+      private: false
+      prompt: |-
+        This playbook will restore a backup of the production server. This will
+        replace the data in many directories. There should be twice the size of
+        this backup available as free disk space, since the archives have to be
+        extracted.
+        Assume you will lose *any existing data* on this server. Are you
+        CERTAIN that you wish to continue? If so, enter "obliteration". Any
+        other value will abort
+
+  tasks:
+    - name: "check that confirmation is given"
+      assert:
+        that:
+          - "confirm == 'obliteration'"
+
+    # We only need this when running on staging, to access
+    # the AWS bucket later that has the production backups
+    - name: "load secrets from production"
+      include_vars:
+        name: "production"
+        file: "../group_vars/production/vault.yml"
+      when: "'production' not in group_names"
+
+    - name: "create temporary directory to store backup archives"
+      tempfile:
+          prefix: "backup-restore-test."
+          state: "directory"
+      register: "backup_restore_tmp_dir"
+
+    - name: "find latest backups"
+      shell:
+        "aws s3 ls s3://sticky-automatic-backups/{{ item }}/
+        | awk -F' ' '{print $NF}'
+        | sort -r
+        | head -n1"
+      register: "backup_filenames"
```

Comment:
What is the type of this filename? Is it a list of strings (one for each invocation of the command)?

Reply:
When using register on a loop, the resulting data is placed into a list within the `results` attribute (see [docs](https://docs.ansible.com/ansible/latest/user_guide/playbooks_loops.html#using-register-with-a-loop)). So `backup_filenames.results[0].stdout` contains the filename of the latest `admins` backup in this case.